frontend/src/container/MetricsApplication/Tabs/Overview.tsx

```diff
@@ -64,12 +84,59 @@ function Application(): JSX.Element {
 		[handleSetTimeStamp],
 	);
 
-	const { topOperations, serviceOverview, topLevelOperations } = useSelector<
-		AppState,
-		MetricReducer
-	>((state) => state.metrics);
+	const serviceOverviewQuery = {
+		queryKey: [servicename, selectedTags, minTime, maxTime],
+		queryFn: (): Promise<PayloadProps> =>
+			getServiceOverview({
+				service: servicename || '',
+				start: minTime,
+				end: maxTime,
+				step: getStep({
+					start: minTime,
+					end: maxTime,
+					inputFormat: 'ns',
+				}),
+				selectedTags,
+			}),
+	};
 
-	const { queries } = useResourceAttribute();
+	const topOperationsQuery = {
+		queryKey: [minTime, maxTime, servicename, selectedTags],
+		queryFn: (): Promise<PayloadPropsTopOpertions> =>
+			getTopOperations({
+				service: servicename || '',
+				start: minTime,
+				end: maxTime,
+				selectedTags,
+			}),
+	};
+
+	const topLevelOperationsQuery = {
+		queryKey: [servicename, minTime, maxTime, selectedTags],
+		queryFn: (): Promise<ServiceDataProps> => getTopLevelOperations(),
+	};
+
+	const queryResult: UseQueryResult[] = useQueries([
+		serviceOverviewQuery,
+		topOperationsQuery,
+		topLevelOperationsQuery,
+	]);
+
+	const serviceOverview = queryResult[0].data as PayloadProps;
+	const serviceOverviewError = queryResult[0].error;
+	const serviceOverviewIsError = queryResult[0].isError;
+	const serviceOverviewIsLoading = queryResult[0].isLoading;
+
+	const topOperations = queryResult[1].data as PayloadPropsTopOpertions;
```

Comment:
make this as component so you don't need to to create for every component

Reply:
This can be done after merging of https://github.com/SigNoz/signoz/pull/3107